packages/jest-util/src/formatFailureMessage.js

```diff
@@ -14,81 +14,131 @@ import type {TestResult} from 'types/TestResult';
 
 const chalk = require('chalk');
 const path = require('path');
+const separateMessageFromStack = require('./separateMessageFromStack');
 
-const ERROR_TITLE_COLOR = chalk.bold.red;
 // filter for noisy stack trace lines
 const JASMINE_IGNORE =
   /^\s+at(?:(?:.*?vendor\/|jasmine\-)|\s+jasmine\.buildExpectationResult)/;
 const STACK_TRACE_IGNORE =
   /^\s+at.*?jest(-.*?)?(\/|\\)(vendor|build|node_modules|packages)(\/|\\)/;
+const MESSAGE_INDENT = '    ';
+const STACK_INDENT = '      ';
+const TITLE_INDENT = '  ';
+const ANCESTRY_SEPARATOR = ' \u203A ';
+const TITLE_BULLET = chalk.bold('\u25cf ');
+const STACK_TRACE_COLOR = chalk.gray;
+const ERROR_MESSAGE_COLOR = chalk.red;
+const STACK_PATH_REGEXP = /\s*at.*\:\d*\:\d*/;
 
-const formatStackTrace = (stackTrace: string, config): string => {
-  const msgBullet = '  - ';
-  const msgIndent = msgBullet.replace(/./g, ' ');
-  let stackTraceLines = 0;
-  return msgBullet + stackTrace
-    // jasmine doesn't give us access to the actual Error object, so we
-    // have to regexp out the message from the stack string in order to
-    // colorize the `message` value
-    .replace(/(^(.|\n)*?(?=\n\s*at\s))/, ERROR_TITLE_COLOR('$1'))
-    .split('\n')
-    .map(line => {
-      // Extract the file path from the trace line.
-      let matches = line.match(/(^\s+at .*?\()([^()]+)(:[0-9]+:[0-9]+\).*$)/);
-      if (!matches) {
-        matches = line.match(/(^\s+at )([^()]+)(:[0-9]+:[0-9]+.*$)/);
-        if (!matches) {
-          return line;
-        }
-      }
-
-      // Filter out noisy and unhelpful lines from the stack trace.
-      // Always keep the first stack trace line (except Jasmine),
-      // even if it comes from Jest.
-      stackTraceLines++;
-      if (
-        config.noStackTrace ||
-        JASMINE_IGNORE.test(line) ||
-        (STACK_TRACE_IGNORE.test(line) && stackTraceLines > 1)
-      ) {
-        return null;
-      }
-
-      const filePath = matches[2];
-      return matches[1] + path.relative(config.rootDir, filePath) + matches[3];
-    })
-    .filter(line => line !== null)
-    .join('\n' + msgIndent);
+const trim = string => string.replace(/^\s+/, '').replace(/\s+$/, '');
+
+// Some errors contain not only line numbers in stack traces
+// e.g. SyntaxErrors can contain snippets of code, and we don't
+// want to trim those, because they may have pointers to the column/character
+// which will get misaligned.
+const trimPaths = string =>
+  string.match(/\s*at.*\:\d*\:\d*/) ? trim(string) : string;
+
+const formatExecError = (testResult: TestResult, config: Config) => {
+  const error = testResult.testExecError;
+  const stack = error.stack
+    ? STACK_TRACE_COLOR(formatStackTrace(error.stack, config))
+    : '';
+  const message = error.stack
+    ? ''
+    : MESSAGE_INDENT + ERROR_MESSAGE_COLOR(error.message) + '\n';
+  return TITLE_INDENT + ERROR_MESSAGE_COLOR('Runtime Error') + '\n' +
```

Comment:
This `Runtime Error` is practically the error saying that `none of the test ran, because something blew up before we even started the suite`

it can be:
1. a run time error. can be: `throw new Error('123')` outside of an `it` block, `required module not found`, etc.
2. `SyntaxError`, when there is a syntax error and `babel` or `vm.Script` fails to parse the source

This type of error is fundamentally different from what `jasmine/mocha` have.
- In jasmine and mocha, this error will break the entire test suite, because every test file is executed in the same process, and none of the tests will ever even run. 
- In jest, it will fail only the current test file and won't affect other files, because everything is isolated.

The message `Runtime Error` doesn't really make sense when it's a SyntaxError.
Should there be a better message? for example: `Failed to run tests`?

current error:
![screen shot 2016-08-08 at 1 55 08 pm](https://cloud.githubusercontent.com/assets/940133/17495944/ebebf62c-5d6f-11e6-9ddc-8aa06b6ad201.png)